inc/Engine/Admin/Settings/Page.php

```diff
@@ -1685,99 +1673,15 @@ private function heartbeat_section() {
-			return;
-		}
-
-		$this->settings->add_page_section(
-			'rocket_insights',
-			[
-				'title'            => __( 'Rocket Insights', 'rocket' ),
-				'menu_description' => __( 'Get performance insights', 'rocket' ),
-			]
-		);
-
-		$this->settings->add_settings_sections(
-			[
-				'performance_monitoring' => [
-					'title'  => __( 'Settings', 'rocket' ),
-					'help'   => [
-						'id'  => '',
-						'url' => '',
-					],
-					'page'   => 'rocket_insights',
-					'helper' => '',
-					'class'  => [ 'pma-settings-container' ],
-				],
-			]
-		);
-
-		/**
-		 * Filters whether to enabled Rocket Insights settings.
-		 *
-		 * @param bool $enabled True to enable settings, false to disable it.
-		 */
-		$insights_settings_enabled = wpm_apply_filters_typed( 'boolean', 'rocket_insights_settings_enabled', true );
-
-		$this->settings->add_settings_fields(
-			[
-				'performance_monitoring' => [
-					'type'              => 'checkbox',
-					'label'             => __( 'Performance Monitoring', 'rocket' ),
-					'description'       => __( 'Enable automatic performance testing for your pages.', 'rocket' ),
-					'section'           => 'performance_monitoring',
-					'page'              => 'rocket_insights',
-					'default'           => 0,
-					'sanitize_callback' => 'sanitize_checkbox',
-					'container_class'   => [
-						! $insights_settings_enabled ? 'wpr-isDisabled' : '',
-					],
-					'input_attr'        => [
-						'disabled' => ! $insights_settings_enabled ? 1 : 0,
-					],
-					'tooltip'           => ! $insights_settings_enabled ? __( 'Upgrade your plan to get access to automatic performance tests', 'rocket' ) : '',
-				],
-				'performance_monitoring_schedule_frequency' => [
-					'container_class'   => [
-						'wpr-field--children',
-						! $insights_settings_enabled ? 'wpr-isDisabled' : '',
-					],
-					'type'              => 'select',
-					'label'             => '',
-					'description'       => '',
-					'parent'            => 'performance_monitoring',
-					'section'           => 'performance_monitoring',
-					'page'              => 'rocket_insights',
-					'default'           => 'monthly',
-					'sanitize_callback' => 'sanitize_text_field',
-					'choices'           => [
-						'daily'   => __( 'Daily', 'rocket' ),
-						'weekly'  => __( 'Weekly', 'rocket' ),
-						'monthly' => __( 'Monthly', 'rocket' ),
-					],
-					'input_attr'        => [
-						'disabled' => ! $insights_settings_enabled ? 1 : 0,
-					],
-				],
-			],
-		);
-	}
-
 	/**
 	 * Registers Add-ons section.
 	 *
 	 * @since 3.0
 	 */
 	private function addons_section() {
-		$webp_beacon       = $this->beacon->get_suggest( 'webp' );
-		$user_cache_beacon = $this->beacon->get_suggest( 'user_cache' );
+		$webp_beacon            = $this->beacon->get_suggest( 'webp' );
+		$user_cache_beacon      = $this->beacon->get_suggest( 'user_cache' );
+		$rocket_insights_beacon = $this->beacon->get_suggest( 'rocket_insights' );
```

Comment:
The addition of the Rocket insights beacon suggest should be done in the rocket_insights_section() method instead